app/src/androidTest/java/com/pileproject/drive/execution/BlockProgramLogicTest.java

```diff
@@ -0,0 +1,195 @@
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pileproject.drive.execution;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import com.pileproject.drive.programming.visual.block.BlockBase;
+import com.pileproject.drive.programming.visual.block.selection.SelectionBlock;
+import com.pileproject.drive.programming.visual.block.selection.SelectionEndBlock;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.internal.util.reflection.Whitebox;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+@RunWith(AndroidJUnit4.class)
+public class BlockProgramLogicTest {
+
+    @Mock ExecutionCondition.SelectionResult trueResult;
+    @Mock ExecutionCondition.SelectionResult falseResult;
+
+    @Mock SelectionBlock selectionBlock;
+
+    @Mock BlockBase trueBlock;
+    @Mock BlockBase falseBlock;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+
+        Whitebox.setInternalState(trueResult, "result", true);
+        Whitebox.setInternalState(falseResult, "result", false);
+
+        // NOTE: commented out lines are using final methods
+        // TODO: consider using PowerMockito
+
+        // doReturn(0).when(selectionBlock).getLeft();
+        // doReturn(199).when(selectionBlock).getRight();
+        Whitebox.setInternalState(selectionBlock, "mLeft", 0);
+        Whitebox.setInternalState(selectionBlock, "mRight", 199);
+
+        // doReturn(0).when(trueBlock).getLeft();
+        // doReturn(99).when(trueBlock).getRight();
+        Whitebox.setInternalState(trueBlock, "mLeft", 0);
+        Whitebox.setInternalState(trueBlock, "mRight", 99);
+
+        // doReturn(100).when(falseBlock).getLeft();
+        // doReturn(199).when(falseBlock).getRight();
+        Whitebox.setInternalState(falseBlock, "mLeft", 100);
+        Whitebox.setInternalState(falseBlock, "mRight", 199);
+    }
+
+    @Test
+    public void whenExecuteConditionHasNoSelectionResult_thenExecuteCurrentBlock() throws Exception {
+
+        ExecutionCondition condition = mock(ExecutionCondition.class);
+        doReturn(0).when(condition).sizeOfSelectionResult(); // no selection result
+
+
+        assertTrue(BlockProgramLogic.willCurrentBlockBeExecuted(condition));
+
+
+        verify(condition).sizeOfSelectionResult();  // do call
```

Comment:
I feel these ``verify``s are kind of over-kill.
What we want to ensure is just "if condition has no condition then returns true".
We should not go into implementation too much.